[DOCS] replace failed tserver process/node/vm etc (#4790)
yugabyte/yugabyte-db: docs/content/latest/troubleshoot/cluster/replace_tserver.md

---

> title: Replace failed tserver

Reviewer: ```suggestion
title: Replace a failed YB-TServer
```

> linkTitle: Replace failed tserver

Reviewer: ```suggestion
linkTitle: RReplace a failed YB-TServer
```

> ---
> title: Replace failed tserver
> linkTitle: Replace failed tserver
> description: Replace failed tserver
> aliases:
>   - /troubleshoot/cluster/reaplace-tserver/
>   - /latest/troubleshoot/cluster/replace-tserver/
> menu:
>   latest:
>     parent: troubleshoot-cluster
>     weight: 829
> isTocNested: true
> showAsideToc: true
> ---

Reviewer: ```suggestion
---
title: Replace a failed YB-TServer
headerTitle: Replace a failed YB-TServer
linkTitle: Replace a failed YB-TServer
description: Procedure to replace a failed YB-TServer
aliases:
  - /troubleshoot/cluster/replace-tserver/
  - /latest/troubleshoot/cluster/replace-tserver/
menu:
  latest:
    identifier: replace-failed-tserver
    parent: troubleshoot-cluster
    weight: 829
isTocNested: true
showAsideToc: true
---
```

> Suppose you have a cluster running and now a tserver has failed and needs to be replaced. The procedure to follow in this 

Reviewer: ```suggestion
If you have a YB-TServer in a YugabytDB cluster that has failed and needs to be replaced, follow the steps below.
```

scenario is:

> ## Start new yb-tserver

Reviewer: ```suggestion
## Start the new YB-TServer
```

> Start up a new yb-tserver, make sure it is in the same placement group as the one you are replacing!

Reviewer: ```suggestion
Install and then start a new YB-TServer, making sure it is in the same placement group as the one you are replacing.
```

> Refer to [starting `tserver` servers](../../../../deploy/manual-deployment/start-tservers/) for further parameters and options.

Reviewer: ```suggestion
For details on starting YB-TServers and more options, see [Start YB-TServers](../../../../deploy/manual-deployment/start-tservers/).
```

> ## Blacklist the old yb-tserver
> 
> Blacklist the old yb-tserver: 
> ```bash
> ~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist ADD $OLD_IP:9100
> ```
> 
> Refer to [change_blacklist](../../admin/yb-admin.md#change-blacklist) command for further parameters and options.
> 

Reviewer: ```suggestion
## Blacklist the failed YB-TServer

To blacklist the failed YB-TServer, run the following command:
 
```sh
~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist ADD $OLD_IP:9100
```

For details, see [`change_blacklist`](../../admin/yb-admin.md#change-blacklist) command in the `yb-admin` reference page.

```

> ## Blacklist the old yb-tserver
> 
> Blacklist the old yb-tserver: 
> ```bash
> ~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist ADD $OLD_IP:9100
> ```
> 
> Refer to [change_blacklist](../../admin/yb-admin.md#change-blacklist) command for further parameters and options.
> 

Reviewer: ```suggestion
## Blacklist the failed YB-TServer

To blacklist the failed YB-TServer, run the following command:
 
```sh
~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist ADD $OLD_IP:9100
```

For details, see [`change_blacklist`](../../admin/yb-admin.md#change-blacklist) command in the `yb-admin` reference page.

```

Reviewer: Add an explanation about why you need to blacklist this address.

Author: @stevebang it's explained on the next step that the master is draining the failed yb-tserver

Author: I've committed all requested changes.

> ## Wait for rebalance
> 
> Wait for data to drain from the old yb-tserver and move into the new one:
> ```bash
> ~/master/bin/yb-admin -master_addresses $MASTERS get_load_move_completion 
> ```

Reviewer: ```suggestion
## Wait for the rebalance to complete

Wait for the data to drain from the failed YB-TServer and for the data to be loaded into the new one. You can check for the completion of rebalancing by running the following command:

```sh
~/master/bin/yb-admin -master_addresses $MASTERS get_load_move_completion 
```
```

> {{< note title="Note" >}}
> This will only complete if the data from the old TS has where to go, that's why it is important to start the new TS first, 
> or ensure your remaining TS have enough capacity and are in the correct placement zones.
> {{< /note >}}

Reviewer: ```suggestion
{{< note title="Note" >}}

Loading and rebalancing will complete if only data from the failed YB-TSserver has someplace to be stored. Start the new YB-TServer first, or ensure your remaining YB-TServers have enough capacity and are in the correct placement zones.

{{< /note >}}
```

> Refer to [get_load_move_completion](../../admin/yb-admin.md#get-load-move-completion) command for further parameters and options.

Reviewer: ```suggestion
For details on using this command, see [`get_load_move_completion`](../../admin/yb-admin.md#get-load-move-completion).
```

> ## Kill old tserver
> Once data move is at 100%, you can kill the old TS (eg: stop the TS process, terminate the VM)
> 
> Wait at least `tserver_unresponsive_timeout_ms` (default 60s) for this TS to be marked as dead by the master leader. 
> You can validate this by visiting `$MASTER_LEADER_IP:7000/tablet-servers` and checking for the state of the old TS to say `DEAD`.

Reviewer: ```suggestion
## Kill the failed YB-TServer

When the data move is complete (100%), kill the failed YB-TServer by stopping the `yb-tserver` process, or terminating the VM.

Now, wait for the YB-TServer to be marked as `DEAD` by the YB-Master leader.  The YB-Master leader will mark the server as `DEAD` after not responding for one minute (based on the `tserver_unresponsive_timeout_ms` (default is `60000`).
 
To verify that the failed YB-TServer is dead, open your web browser to `$MASTER_LEADER_IP:7000/tablet-servers` and check the output.
```

> Remove the old TS from the blacklist: 
> 
> ```bash
> ~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist REMOVE node1:9100
> ```

Reviewer: ```suggestion
## Remove the failed YB-TServer from the blacklist 

Now that the replacement YB-TServer is up and running, and loaded with data, remove the address for the failed YB-TServer from the blacklist.

```sh
~/master/bin/yb-admin -master_addresses $MASTERS change_blacklist REMOVE node1:9100
```
```

> ## Planned cluster changes

Reviewer: Remove this section.

---
title: Replace a failed YB-TServer
headerTitle: Replace a failed YB-TServer
linkTitle: Replace a failed YB-TServer
description: Procedure to replace a failed YB-TServer
aliases:
  - /troubleshoot/cluster/replace-tserver/

>   - /latest/troubleshoot/cluster/replace-tserver/

Reviewer: Remove these -- Aliases are not needed for new pages, except when there are tabbed buttons (then discuss with me).

Author: removed
